colocated-broccoli-plugin.js

```diff
@@ -14,6 +15,21 @@ module.exports = class ColocatedTemplateProcessor extends Plugin {
     this.options = options;
   }
 
+  _transformExistingComponent(filePath, jsContents, templateContents) {
+    if (!jsContents.includes('export default')) {
+      let message = `\`${filePath}\` does not contain a \`default export\`. Did you forget to export the component class?`;
+      return `${jsContents}\nthrow new Error(${JSON.stringify(message)});`;
+    }
+
+    // TODO: deal with hygiene?
+    let output = stripIndents`
+      import hbs from 'ember-cli-htmlbars-inline-precompile';
+      const __COLOCATED_TEMPLATE__ = hbs\`${templateContents}\`;
+      ${jsContents}`;
```

Comment:
Does `stripIndents` look at the template string, or does it look at the interpolated string? (i.e. does the value/content of what is inside of `jsContents` affect the stripping?)

Reply:
(this one does not need trailing newline since `jsContents` should already contain one)

Reply:
@chancancode - It uses the full interpolated value _before_ stripping all leading indentation.

Reply:
I'll add tests confirming that we don't muck up all the leading whitespace in the `jsContents` though. Possibly changing this to:

```js
    let output = stripIndents`
      import hbs from 'ember-cli-htmlbars-inline-precompile';
      const __COLOCATED_TEMPLATE__ = hbs\`${templateContents}\`;
      ` + jsContents;
```

Reply:
Yeah in that case I think (?) the current code is not correct, since jsContents likely have lines without leading indentation, I think it would cause the whole thing to not get stripped properly.

Reply:
Yep, confirm. I’ll add a test (that will fail), then fix the impl to my suggested thing above.

Reply:
This is fixed (no longer using any indentation stripping).